README.md

```diff
@@ -63,65 +63,81 @@ You can get the latest release from Maven:
 The following code sample demonstrates how to get started:
 
 1. Instantiate the SDK.
-2. Define the master key provider.
+2. Setup an AWS KMS Keyring.
 3. Encrypt and decrypt data.
 
 ```java
-// This sample code encrypts and then decrypts a string using a KMS CMK.
-// You provide the KMS key ARN and plaintext string as arguments.
+// This sample code encrypts and then decrypts data using an AWS KMS customer master key (CMK).
 package com.amazonaws.crypto.examples;
 
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.Map;
 
 import com.amazonaws.encryptionsdk.AwsCrypto;
-import com.amazonaws.encryptionsdk.CryptoResult;
-import com.amazonaws.encryptionsdk.kms.KmsMasterKey;
-import com.amazonaws.encryptionsdk.kms.KmsMasterKeyProvider;
+import com.amazonaws.encryptionsdk.AwsCryptoResult;
+import com.amazonaws.encryptionsdk.DecryptRequest;
+import com.amazonaws.encryptionsdk.EncryptRequest;
+import com.amazonaws.encryptionsdk.keyrings.Keyring;
+import com.amazonaws.encryptionsdk.keyrings.StandardKeyrings;
+import com.amazonaws.encryptionsdk.kms.AwsKmsCmkId;
 
-public class StringExample {
-    private static String keyArn;
-    private static String data;
+public class BasicEncryptionExample {
+
+    private static final byte[] EXAMPLE_DATA = "Hello World".getBytes(StandardCharsets.UTF_8);
 
     public static void main(final String[] args) {
-        keyArn = args[0];
-        data = args[1];
+        encryptAndDecrypt(AwsKmsCmkId.fromString(args[0]));
+    }
 
-        // Instantiate the SDK
+    static void encryptAndDecrypt(final AwsKmsCmkId keyArn) {
+        // 1. Instantiate the SDK
         final AwsCrypto crypto = new AwsCrypto();
 
-        // Set up the master key provider
-        final KmsMasterKeyProvider prov = new KmsMasterKeyProvider(keyArn);
+        // 2. Instantiate an AWS KMS Keyring, supplying the key ARN as the generator for generating a 
+        //    data key. While using a key ARN is a best practice, for encryption operations it is also
+        //    acceptable to use a CMK alias or an alias ARN.
```

Comment:
While using a key ARN is a best practice, for encryption operations you can also
        //    use an alias name or alias ARN.